src/views/CountryRiskWatch/RiskBarChart/CombinedChart/index.tsx

```diff
@@ -0,0 +1,384 @@
+                                        riskItem?.[monthName],
+                                        riskItem?.hazard_type,
+                                    ),
+                                ),
+                            })) ?? [],
+                        (data) => data.hazard_type,
+                    ),
+                    WF: {
+                        ...wfRiskDataItem,
+                        ...mapToMap(
+                            monthNumberToNameMap,
+                            (_, monthName) => monthName,
+                            (monthName) => riskScoreToCategory(
+                                wfRiskDataItem?.[monthName],
+                                'WF',
+                            ),
+                        ),
+                    },
+                };
+            }
+
+            const rasterDisplacementData = listToMap(
+                riskData?.raster_displacement_data
+                    ?.map(getDataWithTruthyHazardType)
+                    ?.filter(isDefined) ?? [],
+                (datum) => datum.hazard_type,
+            );
+
+            if (isNotDefined(fiRiskDataItem)) {
+                return rasterDisplacementData;
+            }
+
+            return {
+                ...rasterDisplacementData,
+                FI: fiRiskDataItem,
+            };
+        },
+        [riskData, selectedRiskMetricDetail, fiRiskDataItem, wfRiskDataItem],
+    );
+
+    const filteredRiskData = useMemo(
+        () => {
+            if (isNotDefined(selectedHazardType)) {
+                return selectedRiskData;
+            }
+
+            const riskDataItem = selectedRiskData[selectedHazardType];
+
+            return {
+                [selectedHazardType]: riskDataItem,
+            };
+        },
+        [selectedRiskData, selectedHazardType],
+    );
+
+    const maxValue = useMemo(
+        () => {
+            const maxValueList = Object.values(filteredRiskData ?? {}).map(
+                (riskDataItem) => (
+                    getDisplacementForSelectedMonths(
+                        selectedMonths,
+                        riskDataItem,
+                        'max',
+                    ) ?? 0
+                ),
+            );
+
+            return Math.max(...maxValueList);
+        },
+        [filteredRiskData],
+    );
+
+    const yScale = useMemo(
+        () => getScaleFunction(
+            { min: 0, max: maxValue },
+            { min: 0, max: chartBounds.height },
+            { start: chartMargin.top, end: chartMargin.bottom },
+            true,
+            selectedRiskMetricDetail.key === 'riskScore' ? 'linear' : 'cbrt',
+        ),
+        [chartBounds, maxValue, selectedRiskMetricDetail, chartMargin],
+    );
+
+    const yAxisPoints = useMemo(
+        () => {
+            const numYAxisPoints = 6;
+            const diff = maxValue / (numYAxisPoints - 1);
+
+            return Array.from(Array(numYAxisPoints).keys()).map(
```

Comment:
Let's check if we can use Array.fill

Reply:
No, `Array.fill` is only used when we need same value filled throughout the array